pkg/release/release.go

```diff
@@ -270,14 +270,30 @@ next:
 		status.SetStatusPhase(r.hrClient.HelmReleases(hr.Namespace), hr, v1.HelmReleasePhaseSucceeded)
 		logger.Log("info", "no changes", "phase", action)
 	case InstallAction:
+		if status.IsFailed(hr) {
+			if !status.HasSynced(hr) {
+				// We are in a failed state and the release not been synced, so
+				// we want to reset the install failed counter so installation will continue
+				status.SetInstallPhase(r.hrClient.HelmReleases(hr.Namespace), hr, v1.HelmReleasePhaseInstalling)
+				break
+			}
+			if !status.ShouldRetryInstall(hr, hr.Status.FailedInstallCount) {
+				break
+			}
+		}
+
 		logger.Log("info", "running installation", "phase", action)
 		newRel, err = r.install(client, hr, chart, values)
 		if err != nil {
 			logger.Log("error", err, "phase", action)
 			errs = append(errs, err)
 
-			action = UninstallAction
-			goto next
+			if status.ShouldRetryInstall(hr, hr.Status.FailedInstallCount+1) {
```

Comment:
Just checking on logic: you are passing the count here as it was increased by the unsuccessful installation but the HR does not reflect this?

Reply:
Correct, the way the phase setting works means it will not "actually" be incremented until the next sync loop.